mdsuite/transformations/map_molecules.py

```diff
@@ -119,286 +118,178 @@ def _run_dependency_check(self):
-                List of paths for the dictionary
-        dimension : int
-                Dimension of the property
+                List of paths for which the type species must be built.
+
         Returns
         -------
-        type dict : dict
-                Dictionary for the type spec.
+        type_spec : dict
         """
+        type_spec = {}
         for item in path_list:
-            species = item.split("/")[0]
-            n_atoms = self.experiment.species[species].n_particles
-            dictionary[str.encode(item)] = tf.TensorSpec(
-                shape=(n_atoms, None, dimension), dtype=tf.float64
+            type_spec[str.encode(item)] = tf.TensorSpec(
+                shape=(None, None, 3), dtype=self.dtype
             )
+        type_spec.update(
+            {
+                str.encode("data_size"): tf.TensorSpec(shape=(), dtype=tf.int32),
+            }
+        )
 
-        return dictionary
+        return type_spec
 
-    def _build_reference_graphs(self):
-        """
-        Build the reference graphs from the SMILES strings.
-        Returns
-        -------
-        Nothing.
+    def _build_reduced_mass_dict(self, species: dict, molecular_mass) -> dict:
         """
-        for item in self.molecules:
-            self.reference_molecules[item] = {}
-            mol, species = MolecularGraph(
-                self.experiment,
-                from_smiles=True,
-                smiles_string=self.molecules[item]["smiles"],
-            ).build_smiles_graph()
-            self.reference_molecules[item]["species"] = list(species)
-            self.reference_molecules[item]["graph"] = mol
-            self.reference_molecules[item]["mass"] = self._get_molecular_mass(species)
-
-    def _get_molecular_mass(self, species_dict: dict) -> float:
-        """
-        Get the mass of a SMILES molecule based on experiment data.
+        Build the reduced mass dictionary.
+
+        This is a dictionary of reduced masses for each species in a molecule,
+        i.e m_species / m_molecule. These are used in the COM positions computation
+        later.
 
         Parameters
         ----------
-        species_dict : dict
-                Dictionary of species information along with number of species
-                present in the system.
+        species : list
+                List of species to include in the dict.
+
         Returns
         -------
-        mass : float
-                mass of the molecule
+        reduced_mass_dict : dict
+                Dictionary of reduced masses for each species.
         """
-        mass = 0.0
-        for item in species_dict:
-            mass += self.experiment.species[item]["mass"][0] * species_dict[item]
+        reduced_mass_dict = {}
+        for item in species:
+            reduced_mass_dict[item] = (
+                self.experiment.species[item]["mass"][0] / molecular_mass
+            )
 
-        return mass
+        return reduced_mass_dict
```

Comment:
Same as above, this one should be called `get_reduce_mass_dict` or what I would prefer is `reduce_mass_dict` as a property. Could also be a single line, but dics are less well readable `{species: self.experiment.species[species]["mass"][0] / molecular_mass for species in self.species}`

Reply:
Adapted in 98b8619